math/binary_exponentiation.c

```diff
@@ -0,0 +1,65 @@
+/**
+ * @file
+ * @brief Calculates the nth power of a number in O(log(n)) time.
+ * @details
+ * Calculating the nth power of a number requires O(n) time using the naive
+ * method. This algorithm optimizes it through exponentiation by squaring.
+ * https://cp-algorithms.com/algebra/binary-exp.html
+ * @author [Praneeth Jain](https://github.com/PraneethJain)
+ */
+
+#include <assert.h>  // for assert()
+#include <stdint.h>  // for int64 types
+#include <stdio.h>   // for output
+#include <stdlib.h>  // for exit()
```

Comment:
```suggestion
#include <assert.h>  /// for assert()
#include <stdint.h>  /// for int64 types
#include <stdio.h>   /// for output
#include <stdlib.h>  /// for exit()
```

Reply:
doxygen needs thriple slashes to recognize this as a doc string for the includes